tests/existing/sample/project/src/CMakeLists.txt.after_grouped_update

```diff
@@ -0,0 +1,85 @@
+set(sources
+    # sources
+    project/source1.c
+    project/source2.cc
+    project/source3.cpp
+    project/source4.cxx
+    project/source5.c++
+
+    project/sub2/source1.c
+    project/sub3/source1.c
+
+    project/subfolder/source1.c
+    project/subfolder/source2.cc
+    project/subfolder/source3.cpp
+    project/subfolder/source4.cxx
+    project/subfolder/source5.c++
+    )
+
+set(headers
+    # headers
+    project/header1.h
+    project/subfolder/header4.hxx)
+
+add_library(my_project_lib
+    # sources
+    project/source1.c
+    project/source2.cc
+    project/source3.cpp
+    project/source4.cxx
+    project/source5.c++
+
+    project/sub2/source1.c
+    project/sub3/source1.c
```

Comment:
I'm not sure if grouping just because the dirname has the same length is the best option. You'll get groupings where the dirnames are incidentally the same length but otherwise unrelated.